android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/DVCLifecycleCallbacks.kt

```diff
@@ -0,0 +1,70 @@
+package com.devcycle.sdk.android.eventsource
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import android.os.Handler
+import android.os.HandlerThread
+import timber.log.Timber
+
+class DVCLifecycleCallbacks(eventSource: EventSource, customHandler: Handler? = null) : Application.ActivityLifecycleCallbacks {
+    private val CHECK_DELAY: Long = 800
+
+    private val eventSource = eventSource
+    private val listenerThread: HandlerThread = HandlerThread("DVCForegroundListener")
+    private var handler: Handler? = customHandler
+    private var paused = false
+    private var foreground = true
+    private var check: Runnable? = null
+
+    init {
+        if (handler == null) {
+            listenerThread.start()
+            handler = Handler(listenerThread.looper)
+        }
+    }
+    override fun onActivityResumed(activity: Activity) {
+        paused = false
+        if (check != null) {
+            handler!!.removeCallbacks(check!!) // stop check from running if the timer is still running
+            check = null
+        }
+        if (!foreground) {
+            Timber.d("Restarting connection")
+            eventSource.start()
+        }
+        foreground = true
+    }
+
+    override fun onActivityPaused(activity: Activity) {
+        paused = true
+
+        if (check != null) {
+            handler!!.removeCallbacks(check!!)
```

Comment:
you shouldn't need the non-null assertion on `check` here

Reply:
not sure how to remove it?  the check object will only be non-null if pause was just called and the delay timer is currently running, and removeCallbacks only accepts a non-null value. so i have to check if it's null before calling removeCallbacks (and kotlin is making me re-assert that it's not null in with the '!!' on that line)

Reply:
yeah I thought that once you were inside the `if` statement it would know its not null, but I think because its a `var` it can be changed to null at any time which is why it makes you assert here again